components/menu/menu.tsx

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ReactNode, useState, MouseEvent} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import MuiMenu, {MenuProps as MuiMenuProps} from '@mui/material/Menu';
+import MuiMenuList from '@mui/material/MenuList';
+import {styled as muiStyled} from '@mui/material/styles';
+
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import {getIsMobileView} from 'selectors/views/browser';
+
+import {openModal, closeModal} from 'actions/views/modals';
+
+import CompassDesignProvider from 'components/compass_design_provider';
+import Tooltip from 'components/tooltip';
+import OverlayTrigger from 'components/overlay_trigger';
+import GenericModal from 'components/generic_modal';
+
+const OVERLAY_TIME_DELAY = 500;
+
+interface Props {
+
+    // Trigger button props
+    triggerId?: string;
+    triggerElement: ReactNode;
+    triggerClassName?: string;
+    triggerAriaLabel?: string;
+
+    // Tooltip of Trigger button props
+    triggerTooltipPlacement?: 'top' | 'bottom' | 'left' | 'right';
+    triggerTooltipId?: string;
+    triggerTooltipText?: string;
+    triggerTooltipClassName?: string;
+
+    // Menu props
+    menuId: string;
+    menuAriaLabel?: string;
+
+    children: ReactNode[];
+}
+
+export function Menu(props: Props) {
+    const theme = useSelector(getTheme);
+
+    const isMobileView = useSelector(getIsMobileView);
+
+    const dispatch = useDispatch();
+
+    const [anchorElement, setAnchorElement] = useState<null | HTMLElement>(null);
+    const isMenuOpen = Boolean(anchorElement);
+
+    if (isMobileView) {
+        function MenuModalComponent() {
```

Comment:
I think we might want to separate this out into its own function for organization's sake. It can still live in this file though

Reply:
separated out to its own component